src/apps/bridge/sensor_drivers/aanderaaSensor.cpp

```diff
@@ -53,22 +56,15 @@ void AanderaaSensor::aanderaSubCallback(uint64_t node_id, const char *topic, uin
         uint64_t sensor_reading_time_sec = d.header.sensor_reading_time_ms / 1000U;
         uint32_t sensor_reading_time_millis = d.header.sensor_reading_time_ms % 1000U;
 
-        // Keep track of when we last got a reading from this node
-        // If it is the first reading, or if it has been more than the current aggregation period + 30 seconds
-        // then we will update the node position.
-        static int8_t node_position = 0;
-        static uint32_t last_timestamp = 0;
-        static uint32_t current_timestamp = 0;
-
-        current_timestamp = pdTICKS_TO_MS(xTaskGetTickCount());
-        if ((current_timestamp - last_timestamp > aanderaa->current_agg_period_ms + 30000u) ||
+        uint32_t current_timestamp = pdTICKS_TO_MS(xTaskGetTickCount());
+        if ((current_timestamp - aanderaa->last_timestamp > DEFAULT_AANDERAA_READING_PERIOD_MS + 30000u) ||
```

Comment:
Another fix is now we are using the correct period for comparison. Previously we were using `aanderaa->current_agg_period_ms` but this is equal to the `sampleDuration` not the individual reading period. Since this code runs on the bridge I created a #define for the comparison, but sometime in the future we should get this value from the mote itself.